runtime/src/accounts.rs

```diff
@@ -584,27 +619,31 @@ impl Accounts {
                 );
             }
             let credit = lock.credits.load(Ordering::Relaxed);
-            if credit > 0 {
+            if credit > 0 || lock.rent_debtor.load(Ordering::Relaxed) {
```

Comment:
As I wrote [above](https://github.com/solana-labs/solana/pull/6433/files#r340234135), I believe this should be written like this (or use a lock...):

```Rust
let credit = lock.credits.load(Ordering::Relaxed);
let rent_debtor = lock.rent_debtor.load(Ordering::Acquire);
if credit > 0 || rent_debtor {
   // ... use the `credit` `rent_debtor` local variables
}
```

For example, with current code, `lock.rent_debtor` can be `true` before `lock.credits` is increased by the other thread. In that case, the local variable `credit` loads the stale value before the increase, resulting in evaporating some lamports, ultimately resulting different blockhash for validators.

Reply:
I think this is a non-issue.  the place where these 2 values is considered is after any multi-threaded access.  the iterator in question comes from a take() on the credit_only_locks.

Reply:
@rob-solana Yeah, I jumped into the conclusion too early. [I see no problem now](https://github.com/solana-labs/solana/pull/6433#discussion_r340277060). Sorry for disrupting you...

Reply:
more eyes more better, keep looking!